service-app/internal/api/client.go

```diff
@@ -1,75 +1,98 @@
 package api
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
-	"net/http"
 	"time"
 
+	"github.com/ministryofjustice/opg-scanning/config"
+	"github.com/ministryofjustice/opg-scanning/internal/httpclient"
+	"github.com/ministryofjustice/opg-scanning/internal/logger"
 	"github.com/ministryofjustice/opg-scanning/internal/types"
 )
 
-func CreateStubCase(url string, set types.BaseSet) (*types.ScannedCaseResponse, error) {
-	var scannedCaseRequest types.ScannedCaseRequest
+type CreateStubCase struct {
+	config     config.Config
+	logger     logger.Logger
+	httpClient *httpclient.HttpClient
+}
+
+func NewCreateStubCase(httpClient *httpclient.HttpClient) *CreateStubCase {
+	return &CreateStubCase{
+		config:     httpClient.Config,
+		logger:     httpClient.Logger,
+		httpClient: httpClient,
+	}
+}
+
+// Determines case type and sends the request to Sirius
+func (s CreateStubCase) CreateStubCase(set types.BaseSet) (*types.ScannedCaseResponse, error) {
+	scannedCaseRequest, err := s.determineCaseRequest(set)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine case type: %w", err)
+	}
+
+	return s.requestCreateScannedCase(scannedCaseRequest)
+}
+
+func (s CreateStubCase) determineCaseRequest(set types.BaseSet) (*types.ScannedCaseRequest, error) {
 	now := time.Now().Format(time.RFC3339)
 
-	if set.Header.CaseNo == "" {
-		// Check for LPA cases
-		for _, doc := range set.Body.Documents {
-			if doc.Type == "LPA002" || doc.Type == "LP1F" || doc.Type == "LP1H" || doc.Type == "LP2" {
-				// Create a new LPA case
-				scannedCaseRequest = types.ScannedCaseRequest{
-					BatchID:     set.Header.Schedule,
-					CaseType:    "lpa",
-					ReceiptDate: set.Header.ScanTime,
-					CreatedDate: now,
-				}
-				break
-			} else if doc.Type == "EP2PG" || doc.Type == "EPA" {
-				// Create a new EPA case
-				scannedCaseRequest = types.ScannedCaseRequest{
-					BatchID:     set.Header.Schedule,
-					CaseType:    "epa",
-					ReceiptDate: set.Header.ScanTime,
-					CreatedDate: now,
-				}
-				break
-			}
-		}
-	} else if set.Header.CaseNo != "" {
-		// Check for COPORD case with CaseNo
-		for _, doc := range set.Body.Documents {
-			if doc.Type == "COPORD" {
-				scannedCaseRequest = types.ScannedCaseRequest{
+	for _, doc := range set.Body.Documents {
+		switch doc.Type {
+		case "LPA002", "LP1F", "LP1H", "LP2":
```

Comment:
It would be good to make these constant/enum, to avoid typos and so we can document them

Reply:
Agreed I'll have to follow that up in SSM-23 though, to avoid conflicts.